### ISSUE TITLE
Make some use of `\sarg` in manual examples

### ISSUE THREAD

**Issue by muzimuzhi** [T-F-S/tcolorbox#256]
`\sarg` is supported since v6.0.2 (see #193) and it is designed to represent an optional star as one of command arguments.

Currently there are two examples that document commands accepting an optional star, but for now such stars are both part of command names, hence visually non optional.

**In two examples for `docCommand` env**

I feel like `\foomakedocSubKey` and `\foomakedocSubKey*` are not a good pair of examples, since the real command `\tcbmakedocSubKey` doesn't have a starred form.

Either the second example for `\foomakedocSubKey*` can be removed or `\foomakedocSubKey(*)` can be replaced by some command that does have a starred form, like `\section`

- Current
  <img width="674" alt="image" src="https://github.com/T-F-S/tcolorbox/assets/6376638/d3f26aa2-1e09-4e05-9b0f-5a73e0b6b2b5">
- Suggested
  <img width="674" alt="image" src="https://github.com/T-F-S/tcolorbox/assets/6376638/84c95193-d222-45d4-974a-9aa5a60d9f1b">
  ```tex
  \begin{dispExample}
  \begin{docCommand}{section}{\sarg\oarg{short title}\marg{title}}
    Starts a section.
  \end{docCommand}
  \end{dispExample}
  \begin{dispExample}
  \begin{docCommand}[color definition=blue]{section}{\sarg\marg{title}}
    Starts an unnumbered section.
  \end{docCommand}
  \end{dispExample}
  ```

**In the example for `docCommands` env**

- Current
  <img width="295" alt="image" src="https://github.com/T-F-S/tcolorbox/assets/6376638/0383543c-34cf-4ab4-8b6b-dc5b2c8a3d31">
- Suggested
  <img width="294" alt="image" src="https://github.com/T-F-S/tcolorbox/assets/6376638/da14d3af-888f-4ce1-99f9-988b262d5391">
  ```tex
  \begin{dispExample}
  \begin{docCommands}[
      doc no index,  %  no index entries for this example
      doc name = newtheorem
    ]
    {
      { doc parameter = \sarg\marg{envname} },
      { doc parameter = \marg{envname}\oarg{numbered within} },
      { doc parameter = \oarg{numbered like}\marg{envname} }
    }
    example
  \end{docCommands}
  \end{dispExample}
  ```

PS: For highlighting, `\sarg` need to be added to `\lstdefinestyle{mydocumentation}{...}` in `tcolorbox.doc.s_main.sty`
```diff
diff --git a/doc/latex/tcolorbox/tcolorbox.doc.s_main.sty b/doc/latex/tcolorbox/tcolorbox.doc.s_main.sty
index 5aaf1b5..f2c5339 100644
--- a/doc/latex/tcolorbox/tcolorbox.doc.s_main.sty
+++ b/doc/latex/tcolorbox/tcolorbox.doc.s_main.sty
@@ -205,6 +205,7 @@
     meta,marg,
     oarg,
     refCom,refEnv,refKey,
+    sarg,
     tcbmakedocSubKey,tcbdocmarginnote,tcbdocnew,tcbdocupdated,
     % skins
     tcbincludegraphics,tcbincludepdf,pdfpages,imagepage,imagepath,imagename,
```

**Comment by T-F-S** [comment]
The `\foomakedocSubKey` example is needed for later reference, but I can replace the `\foomakedocSubKey*` by a `\section*` example as suggested. Also, I will change the `newtheorem` example. Thank you.

**Comment by muzimuzhi** [comment]
resolved in eb19e7e (6.2.0 (Preview 3), 2023-11-24).

**Comment by T-F-S** [comment]
Done with https://github.com/T-F-S/tcolorbox/releases/tag/v6.2.0
